src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -244,11 +244,24 @@ public override void Initialize(AnalysisContext context)
 
                             if (additionalLocations.Length != 0 && !AvoidComparing(aooreParameter!))
                             {
-                                context.ReportDiagnostic(condition.CreateDiagnostic(
-                                    UseArgumentOutOfRangeExceptionThrowIfRule,
-                                    additionalLocations,
-                                    properties: ImmutableDictionary<string, string?>.Empty.Add(MethodNamePropertyKey, methodName),
-                                    args: new object[] { nameof(ArgumentOutOfRangeException), methodName! }));
+                                switch (methodName)
+                                {
+                                    case "ThrowIfZero" when aooreThrowIfZero is not null:
+                                    case "ThrowIfNegative" when aooreThrowIfNegative is not null:
+                                    case "ThrowIfNegativeOrZero" when aooreThrowIfNegativeOrZero is not null:
+                                    case "ThrowIfGreaterThan" when aooreThrowIfGreaterThan is not null:
+                                    case "ThrowIfGreaterThanOrEqual" when aooreThrowIfGreaterThanOrEqual is not null:
+                                    case "ThrowIfLessThan" when aooreThrowIfLessThan is not null:
+                                    case "ThrowIfLessThanOrEqual" when aooreThrowIfLessThanOrEqual is not null:
+                                    case "ThrowIfEqual" when aooreThrowIfEqual is not null:
+                                    case "ThrowIfNotEqual" when aooreThrowIfNotEqual is not null:
```

Comment:
NIT: not sure if it is optimized by the compiler, else we could move the strings that used twice into static readonly fields